x/logic/predicate/file.go

```diff
@@ -57,6 +61,62 @@ func SourceFile(vm *engine.VM, file engine.Term, cont engine.Cont, env *engine.E
 	return engine.Delay(promises...)
 }
 
+func Consult(vm *engine.VM, files engine.Term, k engine.Cont, env *engine.Env) *engine.Promise {
+	var filenames []engine.Term
+	iter := engine.ListIterator{List: files, Env: env}
+	for iter.Next() {
+		filenames = append(filenames, iter.Current())
+	}
+	if err := iter.Err(); err != nil {
+		filenames = []engine.Term{files}
+	}
```

Comment:
I saw that the original implementation doesn't handle the error, but I don't see any reason justifying this choice.

Maybe it'd better to avoid silencing it, what do you think?

Reply:
@amimart In fact, the error is handled (`err != nil` 😜) but it's because the `files` term can be a list or a string, in case is a string, iterator throw an error and it's transformed into a list of single filename.

Reply:
Oh I see, the error exists when we have a string instead of a list. This is ugly!